job-dsl-plugin/src/main/groovy/javaposse/jobdsl/plugin/ExecuteDslScripts.java

```diff
@@ -303,7 +317,9 @@ public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher,
         }
 
         if (descriptorMutated) {
-            descriptor.save();
+            if (!isDryRun()) {
+                descriptor.save();
```

Comment:
Not saving the descriptor is not enough, the descriptor must not be modified at all in a dry run. Saving the descriptor just stores state to disk to survive a restart, but memory state must not be touched as well.